lib/streamlit/elements/policies.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import streamlit
+from streamlit.config import get_option
+from streamlit.elements.form import is_in_form
+from streamlit.errors import StreamlitAPIException, StreamlitAPIWarning
+from streamlit.runtime.scriptrunner.script_run_context import get_script_run_ctx
+from streamlit.runtime.state import WidgetCallback, get_session_state
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+
+def check_callback_rules(dg: DeltaGenerator, on_change: WidgetCallback | None) -> None:
+    if streamlit.runtime.exists() and is_in_form(dg) and on_change is not None:
+        raise StreamlitAPIException(
+            "With forms, callbacks can only be defined on the `st.form_submit_button`."
+            " Defining callbacks on other widgets inside a form is not allowed."
+        )
+
+
+_shown_default_value_warning: bool = False
+
+SESSION_STATE_WRITES_NOT_ALLOWED_ERROR_TEXT = """
+Values for st.button, st.download_button, st.file_uploader, st.data_editor,
+st.chat_input, and st.form cannot be set using st.session_state.
+"""
+
+
+def check_session_state_rules(
```

Comment:
nit: maybe it's a good moment to add a small docstring to this method explaining what it does + some documentation for the paramters

Reply:
I have added brief docstrings to the two functions that don't have one. What do you think?